creativecloud/blocks/interactive-metadata/interactive-metadata.js

```diff
@@ -0,0 +1,306 @@
+  if (!intEnb) return null;
+  await intEnbReendered(intEnb);
+  intEnb.classList.add('interactive-enabled');
+  const assets = intEnb.querySelectorAll('.asset picture, .image picture');
+  const iArea = createTag('div', { class: 'interactive-holder show-image' });
+  const pic = assets[assets.length - 1];
+  const newPic = pic.cloneNode(true);
+  const p = createTag('p', {}, newPic);
+  el.querySelector(':scope > div > div').prepend(p);
+  pic.querySelector('img').src = getImgSrc(pic);
+  [...pic.querySelectorAll('source')].forEach((s) => s.remove());
+  const video = createTag('video');
+  const assetArea = intEnb.querySelector('.asset, .image');
+  const container = pic.closest('p');
+  iArea.append(pic, video);
+  if (container) container.replaceWith(iArea);
+  else assetArea.append(iArea);
+  if (intEnb.classList.contains('heading-top')) {
+    const h = intEnb.querySelector('.text').querySelector('h1, h2, h3, h4, h5, h6');
+    if (h) {
+      const htxt = h.textContent;
+      const hTxtTop = createTag('div', { class: 'mobile-top-title' }, htxt);
+      intEnb.querySelector('.image').prepend(hTxtTop);
+    }
+  }
+  const enticementArrow = assetArea.querySelector(':scope > p img[src*="svg"]');
+  if (enticementArrow) {
+    const entP = enticementArrow.closest('p');
+    const entTxt = createTag('div', { class: 'enticement-message' }, entP.textContent);
+    const enticement = createTag('div', { class: 'enticement-container' });
+    enticementArrow.classList.add('enticement-svg');
+    enticement.append(entTxt, enticementArrow);
+    entP.replaceWith(enticement);
+  }
+  return iArea;
+}
+
+function removeAnimation(ia) {
+  const btn = ia.querySelector('.layer .gray-button');
+  if (!btn) return;
+  btn.querySelector('.ia-circle').style.animation = 'none';
+  btn.style.animation = 'none';
+}
+
+async function addBtnAnimation(ia) {
+  const miloLibs = getLibs('/libs');
+  const { createTag } = await import(`${miloLibs}/utils/utils.js`);
+  const btns = ia.querySelectorAll('.layer .gray-button');
+  [...btns].forEach((btn) => {
+    const circle = createTag('div', { class: 'ia-circle' });
+    btn.append(circle);
+    btn.style.animation = 'outline-fill 2700ms 500ms forwards 7';
+    circle.style.animation = 'circle 2700ms ease-in-out 500ms backwards 7';
+    btn.addEventListener('mouseover', () => {
+      removeAnimation(ia);
+    });
+  });
+}
+
+function addAnimationToLayer(ia) {
+  const btn = ia.querySelector('.layer .gray-button');
+  if (btn) addBtnAnimation(ia);
+}
+
+async function renderLayer(stepInfo) {
+  let pResolve = null;
+  let pReject = null;
+  stepInfo.openForExecution = new Promise((resolve, reject) => {
+    pResolve = resolve;
+    pReject = reject;
+  });
+  try {
+    stepInfo.stepIndex = getNextStepIndex(stepInfo);
+    if (stepInfo.stepIndex === 0) stepInfo.displayPath = 0;
+    stepInfo.stepName = stepInfo.stepList[stepInfo.stepIndex];
+    await implementWorkflow(stepInfo);
+    pResolve();
```

Comment:
I believe there should be a better way to handle this instead of leaking resolve and reject.

Reply:
@amitbikram Made the suggested changes